pkg/appmanager/appManager.go

```diff
@@ -1523,7 +1524,8 @@ func (appMgr *Manager) handleConfigForType(
 	found := false
 	plIdx := 0
 	for i, pl := range rsCfg.Pools {
-		if pl.ServiceName == sKey.ServiceName {
+		if pl.ServiceName == sKey.ServiceName &&
+			strings.Contains(pl.Name, sKey.Namespace) {
```

Comment:
I'm concerned about using string.Contains() here, as we can have subtle bugs depending on how users name their namespaces and routes (for example a route name that contains a string that is also the name of a namespace). It would be more accurate to decompose the pool name based on how we format it to correctly identify the namespace embedded in the pool name.

Reply:
I have a semi-solution for this. The reason it's a little tougher is because Route pools are named differently than ConfigMap/Ingress pools. ConfigMap/Ingress pool names start with the namespace. Route pool names start with "openshift", followed by the namespace. To know which case I'm in, I can check if the pool name starts with "openshift", which I'll assume means we are dealing with a Route pool. The only caveat is if there is a ConfigMap/Ingress resource in a namespace called "openshift", it will fall into this condition, which we don't want.

Reply:
I think that is better than using string.Contains(). We should address the naming of pools for ConfigMap/Ingress for consistency and to avoid this problem.

Reply:
Maybe we can address this in your restructure. I think changing this now will cause a lot of issues.